mimic/core.py

```diff
@@ -51,9 +52,25 @@ def __init__(self, clock, apis, domains=()):
         self.domains = list(domains)
 
         for api in apis:
-            this_api_id = ((api.__class__.__name__) + '-' +
-                           random_hex_generator(3))
-            self._uuid_to_api[this_api_id] = api
+            # Gate check the API to make sure it implements one of the
+            # supported interfaces
+            required_interfaces = (IAPIMock, IExternalAPIMock)
+            provided_interfaces = list(zope.interface.providedBy(api))
+            if True in [i in provided_interfaces
```

Comment:
So, this is just a general stylistic comment, and you can do with it what you like, it's definitely not a blocker.

`zope.interface` has some really interesting calculus to decide whether an interface is provided or not, and it's generally best to let it do its job.  As far as I know this implementation is correct, but I always end up forgetting one or two weird edge cases.

But what you're really trying to do here, if you implement it yourself, is to decide if `provided_interfaces` and `required_interfaces` have a non-empty common subset, which can be expressed more gracefully as:

```python
required_interfaces = set([IAPIMock, IExternalAPIMock])
provided_interfaces = set(zope.interface.providedBy(api))
if required_interfaces & provided_interfaces:
    # ...
```

But: better yet would be to avoid implementing implementation-calculus at all on your own, even if it is probably undeniably correct, and instead express this as:

```python
if IExternalAPIMock.providedBy(api):
    this_api_id = ((api.name_key) + '-' + random_hex_generator(3))
elif IAPIMock.providedBy(api):
    this_api_id = ((api.__class__.__name__) + '-' + random_hex_generator(3))
else:
    raise TypeError(
        api.__class__.__module__ + '/' +
        api.__class__.__name__ +
        " does not implement IAPIMock or IExternalAPIMock"
    )
self._uuid_to_api[this_api_id] = api
```